stacks-signer/src/main.rs

```diff
@@ -169,6 +209,8 @@ fn main() {
         )
     };
 
+    let (_cmd_send, cmd_recv) = channel();
+    let (res_send, _res_recv) = channel();
```

Comment:
You should update the DKG and SIGN CLI commands to use this res_recv so that they terminate upon completion.